modules/apps/object/object-test/src/testFunctional/tests/CreateObject.testcase

```diff
@@ -2548,6 +2548,62 @@ definition {
 		CreateObject.assertCustomObjectNotPresent(labelName = "Custom Object 71");
 	}
 
+	@description = "LPS-151877 - Verify that, when adding a new Object, the admin user is able to restrict users to only see entries from an account that they are part of"
+	@priority = "5"
+	test CanOnlySeeEntriesFromAccount {
+		property portal.acceptance = "true";
+
+		ObjectAdmin.addObjectViaAPI(
+			labelName = "Custom Object 156822A",
+			objectName = "CustomObject156822A",
+			pluralLabelName = "Custom Objects 156822A");
+
+		ObjectAdmin.addObjectViaAPI(
+			labelName = "Custom Object 156822B",
+			objectName = "CustomObject156822B",
+			pluralLabelName = "Custom Objects 156822B");
+
+		ObjectAdmin.addObjectRelationshipViaAPI(
+			objectName_1 = "AccountEntry",
+			objectName_2 = "CustomObject156822A",
+			relationshipLabel = "RelationshipAccount",
+			relationshipName = "relationshipAccount",
+			relationshipType = "oneToMany");
+
+		ObjectAdmin.addObjectRelationshipViaAPI(
+			objectName_1 = "CommerceOrder",
+			objectName_2 = "CustomObject156822A",
+			relationshipLabel = "RelationshipCommerceOrder",
+			relationshipName = "relationshipCommerceOrder",
+			relationshipType = "oneToMany");
+
+		ObjectAdmin.addObjectRelationshipViaAPI(
+			objectName_1 = "User",
+			objectName_2 = "CustomObject156822A",
+			relationshipLabel = "RelationshipUser",
+			relationshipName = "relationshipUser",
+			relationshipType = "oneToMany");
+
+		ObjectAdmin.addObjectRelationshipViaAPI(
+			objectName_1 = "CustomObject156822B",
+			objectName_2 = "CustomObject156822A",
+			relationshipLabel = "CustomObject156822B",
+			relationshipName = "customObject156822B",
+			relationshipType = "oneToMany");
+
+		ObjectAdmin.openObjectAdmin();
+
+		ObjectPortlet.selectCustomObject(label = "Custom Object 156822");
+
+		CreateObject.activeToogleAccountRestriction();
+
+		for (var restrictedFields : list "RelationshipCommerceOrder,RelationshipUser,CustomObject156822B") {
+			CreateObject.assertAccountRestrictedFieldNotPresent(accountRestrictedFieldNotPresent = "${restrictedFields}");
+		}
+
+		CreateObject.selectAccountRestrictedField(accountRestrictedField = "RelationshipAccount");
+	}
+
```

Comment:
For this test, we should be testing two things: 
- Assert users who are part of the account see the restricted entries
- Assert users who aren't part of the account cannot see the restricted entries

So here, don't need to assert the field isn't present since that's not what's important. We need to create the object entries as well as a couple users to check whether or not the restriction works correctly.

Reply:
Also, the user that is part of the acount should be able to add more entries. The user that isn't part of the account shouldn't be able to add entries, according to the acceptance criteria.